$OpenBSD: patch-src_main_c,v 1.4 2010/03/29 12:16:07 dcoppa Exp $

fix from upstream svn:
fix memory leak when we have an invalid item to play

set some sane defaults:
use mplayer's software volume control (see pkg/MESSAGE)
disable bar animation in full screen
use xscrnsaver controls over gnome-power-manager

--- src/main.c.orig	Thu Feb 25 17:01:41 2010
+++ src/main.c	Wed Mar 24 11:08:38 2010
@@ -260,6 +260,8 @@ gint play_iter(GtkTreeIter * playiter, gint restart_se
     } else {
         if (verbose > 1)
             printf("iter is invalid, nothing to play\n");
+        g_free(thread_data);
+        thread_data = NULL;
         return 0;
     }
 
@@ -688,7 +690,7 @@ int main(int argc, char *argv[])
     playlist_visible = FALSE;
     disable_fullscreen = FALSE;
     disable_framedrop = FALSE;
-    softvol = FALSE;
+    softvol = TRUE;
     remember_softvol = FALSE;
     volume_softvol = -1;
     volume_gain = 0;
@@ -703,7 +705,7 @@ int main(int argc, char *argv[])
     reallyverbose = 0;
     embedding_disabled = FALSE;
     disable_pause_on_click = FALSE;
-    disable_animation = FALSE;
+    disable_animation = TRUE;
     auto_hide_timeout = 3;
     use_mediakeys = TRUE;
     use_defaultpl = FALSE;
@@ -742,7 +744,7 @@ int main(int argc, char *argv[])
     start_second = 0;
     play_length = 0;
     save_loc = TRUE;
-    use_xscrnsaver = FALSE;
+    use_xscrnsaver = TRUE;
     screensaver_disabled = FALSE;
 
     sa.sa_handler = hup_handler;
